Fix #5782: Added custom check to remove backslash continuation

ankita240796 opened the pull request:
## Explanation
Fixes #5782: Added custom pylint checker to ensure that backslash is not used for breaking long lines.

## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python scripts/pre_commit_linter.py` and `bash scripts/run_frontend_tests.sh`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [x] The PR is assigned to an appropriate reviewer.

ankita240796 committed:
Added custom check to remove backslash continuation

import-keshav commented:
Hi @ankita240796, please try to convert this code into this type of code.

```
file_content = node.stream().readlines()
file_length = len(file_content)

for line_num in xrange(file_length):
     line = file_content[line_num]
     if line.rstrip('\r\n').endswith('\\'):
         self.add_message('backslash-continuation', line=line_num + 1)
```  

```
with node.stream().readlines() as stream:
      for (line_num, line) in enumerate(stream):
            if line.rstrip('\r\n').endswith('\\'):
                self.add_message('backslash-line-continuation',line=line_num+1)
```
as it will be more pythonic, and less time consuming.
Thanks!

ankita240796 committed:
Address review comments

ankita240796 commented:
Hi @import-keshav

Thanks for the review. I have made the changes, PTAL!

import-keshav commented:
LGTM @ankita240796.
Thanks!

DubeySandeep commented:
Hi @apb7, can you please review/approve this PR? (It needs your approval as a code owner.)

apb7 reviewed (commented):
Hi @ankita240796 , this looks good. Have you tested it? Can you please post a screenshot for the same? Also, for each check in pylint_extensions, we do write a corresponding test in pylint_extensions_test.py: https://github.com/oppia/oppia/blob/develop/scripts/pylint_extensions_test.py.
Can you add the tests for this check as well?

ankita240796 committed:
Add test for backslash checker

ankita240796 commented:
Hi @apb7, I have added test for the custom backslash continuation checker, PTAL!

Here are the screenshots for testing the checker:

I added custom messages in one of the files as follows:

![image](https://user-images.githubusercontent.com/15226041/50040449-84bc2680-0069-11e9-8a34-8eb652c47a94.png)

Then I ran the linter and the output was as expected:

![image](https://user-images.githubusercontent.com/15226041/50040444-75d57400-0069-11e9-93f4-a2eaf565831a.png)

apb7 commented on scripts/pylint_extensions_test.py:
```diff
@@ -189,3 +189,44 @@ def test_finds_import_from(self):
+        node = astroid.scoped_nodes.Module(name='test', doc='Custom test')
+        temp_file = tempfile.NamedTemporaryFile()
+        filename = temp_file.name
+        # pylint: disable=backslash-continuation
```
Do you need to disable the rule for this complete block? I think disabling for a single line would be suffice.

ankita240796 replied:
Yes, fixed this. PTAL!

ankita240796 committed:
Add comment for single lines

apb7 reviewed (approved):
LGTM, thanks @ankita240796!
